pkg/catalog/catalog.go

```diff
@@ -431,6 +439,14 @@ func New(ctx context.Context, cfg Config) (*Catalog, error) {
 		addressProvider:       addressProvider,
 		deleteSensor:          deleteSensor,
 		signingKey:            cfg.Config.StorageConfig().SigningKey(),
+		// Initiate the API callback function
+		APIErrorCB: func(ctx context.Context, w http.ResponseWriter, r *http.Request, err error, cb func(w http.ResponseWriter, r *http.Request, code int, v interface{})) bool {
```

Comment:
Previously commented, why do we have a default initialization for this? 
**Is there a scenario where we want ithis default to actually behave that way?** 
If yes, in what case?
If no, then set a dummy noope default that does nothing. 
BTW, The best option is to find a way to make it required during initialization (I know it's difficult since the error callback is only available in controller AFTER the catalog already instansiated, Im ok with this not being done now).